File: C++/seg_model.cpp
#include <string>
#include "ultralytics/YOLO.h"

int main() {
    // Load a model
    // YOLO model("yolo11n-seg.pt");

    // Create YOLO instance
    YOLO model;

    // Training parameters
    TrainingConfig config;
    config.data = "configmodel.yaml";
    config.epochs = 1;
    config.imgsz = 640;

    // Train the model
    Results results = model.train(config);

    // Load trained model
    // YOLO trainedModel("best.pt");

    // Inference
    // Results inferenceResults = trainedModel.predict("D:/Desktop/Adas_test/ds_segmentation/images/img6.jpg");

    return 0;
}

File: seg_model.py
from ultralytics import YOLO

# Load a model
#model = YOLO("yolo11n-seg.pt")

# Train the model
results = model.train(data="configmodel.yaml", epochs=1, imgsz=640)

#model = YOLO("best.pt")


#results = model("D:/Desktop/Adas_test/ds_segmentation/images/img6.jpg")
